### PR TITLE
:bug: Failed to sync sa work-controller-sa in cluster manger hosted mode

### PR DESCRIPTION
**Pull request by zhujian7**
## Summary
1. Fix issue for cluster manger in hosted mode:
```
apiVersion: operator.open-cluster-management.io/v1
kind: ClusterManager
metadata:
  name: cluster-manager
spec:
  addOnManagerImagePullSpec: quay.io/open-cluster-management/addon-manager:latest
  deployOption:
    hosted:
      registrationWebhookConfiguration:
        address: management-control-plane
        port: 30443
      workWebhookConfiguration:
        address: management-control-plane
        port: 31443
    mode: Hosted
  placementImagePullSpec: quay.io/open-cluster-management/placement:latest
  registrationImagePullSpec: quay.io/open-cluster-management/registration:latest
  workImagePullSpec: quay.io/open-cluster-management/work:latest
status:
  conditions:
  - lastTransitionTime: "2023-07-14T07:51:04Z"
    message: 'Failed to sync service account: serviceaccounts "work-controller-sa"
      not found'
    reason: ServiceAccountSyncFailed
    status: "False"
    type: Applied
```

cluster-manager logs:
```
E0714 07:51:54.910585       1 base_controller.go:270] "ClusterManagerController" controller failed to sync "cluster-manager", err: serviceaccounts "work-controller-sa" not found
E0714 07:52:37.108738       1 base_controller.go:270] "ClusterManagerController" controller failed to sync "cluster-manager", err: serviceaccounts "work-controller-sa" not found
E0714 07:54:00.264383       1 base_controller.go:270] "ClusterManagerController" controller failed to sync "cluster-manager", err: serviceaccounts "work-controller-sa" not found
```

2. add permission for cluster manager to get secret `addon-manager-controller-sa-kubeconfig`

3. fix flaky integration test
## Related issue(s)

Fixes #

### CODE REVIEW

File: pkg/operator/operators/clustermanager/controllers/clustermanagercontroller/clustermanager_controller.go [hunk 1]
```diff
@@ -55,7 +55,8 @@ type clusterManagerController struct {
 	cache                resourceapply.ResourceCache
 	// For testcases which don't need these functions, we could set fake funcs
 	ensureSAKubeconfigs func(ctx context.Context, clusterManagerName, clusterManagerNamespace string,
-		hubConfig *rest.Config, hubClient, managementClient kubernetes.Interface, recorder events.Recorder) error
+		hubConfig *rest.Config, hubClient, managementClient kubernetes.Interface, recorder events.Recorder,
+		mwctrEnabled, addonManagerEnabled bool) error
```
Comment (qiujian16) [inline pkg/operator/operators/clustermanager/controllers/clustermanagercontroller/clustermanager_controller.go:59]: can we just use features? it is global variable, we do not need them as input

**Comment by qiujian16** [comment]
also I think we need to enhance the integration test for this bug?

File: pkg/operator/operators/clustermanager/controllers/clustermanagercontroller/clustermanager_controller.go (hunk 1 above)
Comment (zhujian7) [reply pkg/operator/operators/clustermanager/controllers/clustermanagercontroller/clustermanager_controller.go:59]: for hosted mode, there might be multiple clusterManager CRs, if they have different feature gate settings, I think using the global variable is not appropriate.

**Comment by zhujian7** [comment]
cc @xuezhaojun @zhiweiyin318

**Pull request merged**